docs/fobnail-api.md

```diff
@@ -0,0 +1,387 @@
+# Fobnail API
+
+Fobnail Token exposes API over
+[CoAP](https://datatracker.ietf.org/doc/html/rfc7252) server exposed over UDP.
+
+## General considerations
+
+This document describes API version 1 and all API endpoint addresses are
+relative to (starting with) `/api/v1`. Some APIs create temporary objects that
+live at most as long as client is connected. To check connection status server
+issues CoAP Ping requests which clients **must** respond to.
+
+When an object has been created, server responds with **2.01**, `Location-Path`
+specifies path to the resource (a numeric object ID). Object IDs are specific to
+a client - client cannot access object IDs of other clients.
+
+### Error responses
+
+Fobnail Token may return error codes as defined by CoAP specification. An error
+response **may** contain additional context sent as payload. All cacheable error
+responses **must** have `Max-Age` set to 0.
+
+### Fobnail Data format
+
+Fobnail uses CBOR (with a few exceptions) for transmitting structured data both
+from Fobnail to Attester and from Attester to Fobnail. We use subset of
+[RFC7049](https://datatracker.ietf.org/doc/html/rfc7049) (no tagging and no
+extensions).
+
+Clients are **required** to set `Content-Format` to `Application/CBOR` when
+transmitting CBOR-encoded data. Clients **may** set `Accept` header, in that
+case server may return an error as defined
+[here](https://datatracker.ietf.org/doc/html/rfc7252#section-5.10.4).
+
+## API endpoints
+
+### Fobnail Token provisioning
+
+| Endpoint Name             | Method | Arguments                               |
+| ------------------------- | ------ | --------------------------------------- |
+| /admin/token_provision    | POST   | PO certificate chain                    |
+| /admin/provision_complete | POST   | Fobnail Identity/Encryption certificate |
+
+These APIs may be executed only when Fobnail Token is in a unprovisioned state.
+Fobnail Token **must** respond with **4.03** if these APIs are called after
+provisioning is complete.
+
+#### /admin/token_provision
+
+Client transfers PO certificate chain in the same format as used for EK
+certificate chain. Fobnail Token verifies certificate chain against the trust
```

Comment:
EK certificate chain was not described yet. In order to reduce jumping over the document I would move description of format here and link to it in EK.

Reply:
Fixed